src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -676,3 +690,132 @@ def _diskPNNs(self, pnnList):
             else:
                 diskPNNs.add(pnn)
         return diskPNNs
+
+    def updateSites(self, doc):
+        """
+        Update sites API provides asynchronous update of Site info.
+
+        :param doc: JSON payload with the following data structures:
+                    [{'workflow': <wflow name>, 'SiteWhiteList' ['T1', ...], 'SiteBlackList': ['T2',...]}, {...}]
+        :return: acknowledge dict to upstream caller (ReqMgr2)
+        """
+        # preserve provided payload to local file system
+        errors = []
+        for rec in doc:
+            wflow = rec['workflow']
+            status = self.saveData(wflow, rec)
+            if status != 'ok':
+                errors.append({'workflow': wflow, 'error': status})
+        # send acknowledged message back to upstream caller
+        resp = {'status': 'ok'}
+        if len(errors) != 0:
+            resp = {'status': 'fail', 'errors': errors}
```

Comment:
I would suggest to use the same string as we use in CouchDB, just so we keep error strings as consistent as possible. Please check out the CMSCouch.py module, which I believe sets the non `ok` answer.

Reply:
This API is used by HTTP end-point to return to upstream caller. Please clearly define how HTTP end-point should behave both in success and failure mode? In other words, if this API succeed, what it should return, a code , nothing? And if it fails what it should return to upstream code, a string? How error can be defined in upstream code from a return value of this API?

Reply:
And, on a separate note, why MSTransferor or in this sense any MS service should be complaint how CMSCouch return errors? I'm not criticizing but rather trying to understand. Bottom line, I'm asking how any MS service should return the success and failures? Is it standardized across all MS services?

Reply:
I was probably mixing things up and ended up thinking that this data structure was written to couchdb, hence reporting any potential errors from the backend database back to the user.

Seeing that I was wrong, I would suggest you to look into MSPileup (or perhaps pick a different MS service) to see how the server responds back to the client, which data format and content is returned. Just so we try to keep services as consistent as possible.